domain-management/src/main/java/org/jboss/as/domain/management/access/AccessIdentityResourceDefinition.java

```diff
@@ -170,7 +172,7 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
         @Override
         protected boolean requiresRuntime(OperationContext context) {
-            return true;
+            return context.getProcessType() != ProcessType.EMBEDDED_SERVER || context.getRunningMode() != RunningMode.ADMIN_ONLY;
```

Comment:
This needs to account for ProcessType.EMBEDDED_HOST_CONTROLLER as well.

I'm not sure on this but perhaps ProcessType could expose and isEmbedded() property to help with keeping this particular kind of discrimination consistent. My only hesitation is whether that will always have a single value for all process types. For example, for SELF_CONTAINED it's now false but maybe someday there would be a flavor of that kind of thing where it is ambiguous.